test/index.test.ts

```diff
@@ -132,3 +132,20 @@ describe("createApp", () => {
     );
   });
 });
+
+describe("createApp with target domain", () => {
+  it("should create app with target domain", () => {
+    const app = createApp(origin);
+    const state = app?.getState();
+
+    expect(state.domain).toBe(origin);
+  });
+
+  it("should create app without target domain passed", () => {
+    // Defined with "Object.defineProperty"
+    const app = createApp();
+    const state = app?.getState();
+
+    expect(state.domain).toBe("saleor.domain");
+  });
```

Comment:
I think this repo structure is a little too tangled. 

Can we create some smaller files (eg `create-app.ts`) and it's corresponding files for unit tests (`create-app.test.ts`)? 

Then `/test` directory can be integration/e2e

I'd like to avoid having all tests in single file

Reply:
I'm not sure I follow 😛 So you'd like to change what's in `index` to `create-app.ts`? All the tests in index are very similar to each other, I'm not really sure how to 'divide' it properly

Reply:
Okay let's merge it now, but I think we should later refactor it a little.